oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/server/logout/OidcClientInitiatedServerLogoutSuccessHandler.java

```diff
@@ -72,22 +76,11 @@ public OidcClientInitiatedServerLogoutSuccessHandler(
 
 	@Override
 	public Mono<Void> onLogoutSuccess(WebFilterExchange exchange, Authentication authentication) {
+		RedirectUriParameters redirectUriParameters = new RedirectUriParameters();
+		redirectUriParameters.setAuthentication(authentication);
+		redirectUriParameters.setServerWebExchange(exchange.getExchange());
 		// @formatter:off
-		return Mono.just(authentication)
```

Comment:
Can we please look up the `ClientRegistration` before passing to the resolver?

Reply:
Currently, `redirectUriResolver` finds `clientRegistration` in `clientRegistrationRepository`. To do this, it needs to get the `authorizedClientRegistrationId` from `authentication`. It will not be possible to set `clientRegistration` before passing it to `redirectUriResolver`, since by default it has not yet been found at this moment.

Reply:
I don't think I'm following. Why can't the code lookup the registration id from the authentication before looking up the client registration? That's what it was doing before. What am I missing?

IOW, could you instead only have the resolver perform the steps from [here](https://github.com/spring-projects/spring-security/pull/14808/files#diff-9b5866925549763b9729318297077754e6ffa9c14766158fd3ea6fa466cef86dL83) to [here](https://github.com/spring-projects/spring-security/pull/14808/files#diff-9b5866925549763b9729318297077754e6ffa9c14766158fd3ea6fa466cef86dL89)?

Reply:
I understood your idea, maybe I initially went the wrong way. Could you please review the changes?